internal/unionstore/pipelined_memdb.go

```diff
@@ -174,6 +176,62 @@ func (p *PipelinedMemDB) GetFlags(k []byte) (kv.KeyFlags, error) {
 	return f, nil
 }
 
+func (p *PipelinedMemDB) Prefetch(ctx context.Context, keys [][]byte) (map[string][]byte, error) {
+	m := make(map[string][]byte, len(keys))
+	if p.prefetchCache == nil {
+		p.prefetchCache = make(map[string]util.Option[[]byte], len(keys))
+	}
+	shrinkKeys := make([][]byte, 0, len(keys))
+	for _, k := range keys {
+		v, err := p.get(ctx, k, true)
+		if err != nil {
+			if tikverr.IsErrNotFound(err) {
+				shrinkKeys = append(shrinkKeys, k)
+				continue
+			}
+			return nil, err
+		}
+		if len(v) > 0 {
+			m[string(k)] = v
+			p.prefetchCache[string(k)] = util.Some(v)
+		} else {
+			p.prefetchCache[string(k)] = util.None[[]byte]()
+		}
+	}
+	storageValues, err := p.bufferBatchGetter(ctx, shrinkKeys)
```

Comment:
BufferBatchGet doesn't distinguish NotExist and empty value?

Reply:
See the collect function, NotExist won't be in the returned pairs, and empty values are. So when reading value from the map `storageValues`, we can distinguish this case.

https://github.com/ekexium/client-go/blob/925124f5b319d943813f29de30990bfe4572e8ef/txnkv/txnsnapshot/snapshot.go#L257-L266